pkg/collector/corechecks/servicediscovery/module/impl_linux_test.go

```diff
@@ -364,6 +365,78 @@ func buildFakeServer(t *testing.T) string {
 	return filepath.Dir(serverBin)
 }
 
+const (
+	pythonScript = `#! /usr/bin/python
+
+import socket
+import time
+
+HOST = '127.0.0.1'
+PORT = # Empty port
+
+s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+s.bind((HOST, PORT))
+s.listen()
+time.sleep(30)
+`
+)
+
+func TestPythonFromBashScript(t *testing.T) {
+	t.Run("PythonFromBashScript", func(t *testing.T) {
+		testCaptureWrappedCommands(t, "python-*.sh", []byte(pythonScript), []string{"sh", "-c"}, func(service model.Service) bool {
+			return service.Language == string(language.Python)
+		})
+	})
+	t.Run("DirectPythonScript", func(t *testing.T) {
+		testCaptureWrappedCommands(t, "script.py", []byte(pythonScript), nil, func(service model.Service) bool {
+			return service.Language == string(language.Python)
+		})
+	})
+}
+
+func testCaptureWrappedCommands(t *testing.T, scriptName string, scriptContent []byte, commandWrapper []string, validator func(service model.Service) bool) {
+	// Creating tempfile
```

Comment:
What's the advantage of this over just checking in the script in `testdata`?

Reply:
no advantange, changed